test/e2e/case20_delete_objects_test.go

```diff
@@ -142,14 +178,34 @@ var _ = Describe("Test Object deletion", Ordered, func() {
 
 				utils.CheckComplianceStatus(g, managedPlc, "Compliant")
 			}, defaultTimeoutSeconds, 1).Should(Succeed())
+			Eventually(func() bool {
+				managedPlc := utils.GetWithTimeout(clientManagedDynamic, gvrConfigPolicy,
+					case20ConfigPolicyNameInform, testNamespace, true, defaultTimeoutSeconds)
+				relatedObj := managedPlc.Object["status"].(map[string]interface{})["relatedObjects"].([]interface{})[0]
+				properties := relatedObj.(map[string]interface{})["properties"].(map[string]interface{})
+
+				return properties["createdByPolicy"].(bool)
+			}, defaultTimeoutSeconds, 1).Should(BeFalse())
 			Eventually(func() interface{} {
 				managedPlc := utils.GetWithTimeout(clientManagedDynamic, gvrConfigPolicy,
 					case20ConfigPolicyNameInform, testNamespace, true, defaultTimeoutSeconds)
 				relatedObj := managedPlc.Object["status"].(map[string]interface{})["relatedObjects"].([]interface{})[0]
-				properties := relatedObj.(map[string]interface{})["properties"]
+				properties := relatedObj.(map[string]interface{})["properties"].(map[string]interface{})
+
+				return properties["uid"]
+			}, defaultTimeoutSeconds, 1).ShouldNot(BeEmpty())
+			Eventually(func() bool {
+				managedPlc := utils.GetWithTimeout(clientManagedDynamic, gvrConfigPolicy,
+					case20ConfigPolicyNameInform, testNamespace, true, defaultTimeoutSeconds)
+				relatedObj := managedPlc.Object["status"].(map[string]interface{})["relatedObjects"].([]interface{})[0]
+				properties := relatedObj.(map[string]interface{})["properties"].(map[string]interface{})
 
-				return properties
-			}, defaultTimeoutSeconds, 1).Should(BeNil())
+				if dryRunNoOpOverride, exists := properties["dryRunNoOpOverride"]; exists {
+					return dryRunNoOpOverride.(bool)
+				}
+
+				return false
+			}, defaultTimeoutSeconds, 1).Should(BeFalse())
```

Comment:
Is there a test that checks when this new status field is supposed to be `true`?

I guess it might be hard to find a field that triggers this for a test in KinD: the example in the bug is openshift-specific.

In general, I think these `delete_objects_test` scenarios aren't exactly the right spot for verifying this field.

Reply:
Let me see if Kind has any default admission controllers to reproduce this behaviour